metrics/interface/src/main/java/org/apache/iotdb/metrics/metricsets/disk/LinuxDiskMetricsManager.java

```diff
@@ -0,0 +1,397 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.metrics.metricsets.disk;
+
+import org.apache.iotdb.metrics.config.MetricConfigDescriptor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * All data metrics are collected from <b>/proc/</b>.
+ *
+ * <p><b>/proc/diskstats</b> is a file in Linux, it contains the running information of the disks,
+ * including device name, read operation count, merged read operation count, write operation count,
+ * merged write operation count etc. This class collects the data periodically and analyzes the
+ * changes in each pieces of data to gain an insight of the disks of status.
+ *
+ * <p><b>/proc/[PID]/io</b> is also a file in Linux, it indicates the io status of a specific
+ * process. The content of it contains following items: actual read byte, actual write byte, read
+ * system call count, write system call count, byte attempt to read, byte attempt to write,
+ * cancelled write byte.
+ */
+public class LinuxDiskMetricsManager implements IDiskMetricsManager {
+  private final Logger log = LoggerFactory.getLogger(LinuxDiskMetricsManager.class);
+
+  @SuppressWarnings("squid:S1075")
+  private static final String DISK_STATUS_FILE_PATH = "/proc/diskstats";
+
+  @SuppressWarnings("squid:S1075")
+  private static final String DISK_ID_PATH = "/sys/block";
+
+  private final String processIoStatusPath;
+  private static final int DISK_ID_OFFSET = 3;
+  private static final int DISK_READ_COUNT_OFFSET = 4;
+  private static final int DISK_MERGED_READ_COUNT_OFFSET = 5;
+  private static final int DISK_SECTOR_READ_COUNT_OFFSET = 6;
+  private static final int DISK_READ_TIME_OFFSET = 7;
+  private static final int DISK_WRITE_COUNT_OFFSET = 8;
+  private static final int DISK_MERGED_WRITE_COUNT_OFFSET = 9;
+  private static final int DISK_SECTOR_WRITE_COUNT_OFFSET = 10;
+  private static final int DISK_WRITE_TIME_COST_OFFSET = 11;
+  private static final int DISK_QUEUE_SIZE_OFFSET = 12;
+  private static final int DISK_IO_TOTAL_TIME_OFFSET = 13;
+  private static final long UPDATE_SMALLEST_INTERVAL = 10000L;
+  private Set<String> diskIdSet;
+  private long lastUpdateTime = 0L;
+  private long updateInterval = 1L;
+
+  // Disk IO status structure
+  private final Map<String, Long> lastReadOperationCountForDisk = new HashMap<>();
```

Comment:
How about use `getDiskIds().size()` to be the initial capacity of these maps as `DEFAULT_INITIAL_CAPACITY` in HashMap is 16, which is usually bigger than the number of disks